checkov/common/resource_code_logger_filter.py

```diff
@@ -0,0 +1,36 @@
+from __future__ import annotations
+
+import os
+from logging import Logger, Filter, LogRecord
+
+from checkov.common.util.type_forcers import convert_str_to_bool
+
+
+class ResourceCodeFilter(Filter):
+    """
+    A custom logger filter designed to decide if we want to filter some logs from the default logger.
+    Could be used to reduce logs size.
+    First use case is to log without the actual code of resources, which takes a lot of the logs size.
+    The default is to log everything in order to keep api the same.
+    """
+    CODE_TEMPLATES: list[str] = []
+
+    def __init__(self, allow_code_logging: bool = True):
+        super().__init__()
+        self.allow_code_logging = allow_code_logging
+
+    def filter(self, record: LogRecord) -> bool:
+        if self.allow_code_logging:
+            return True
+        msg = record.msg
+        for code_template in ResourceCodeFilter.CODE_TEMPLATES:
```

Comment:
can we check an extra variable that will be sent in those logs? I don't think it will be possible to create a `CODE_TEMPLATES` list that will include all the code log cases

Reply:
what do you call an extra variable? basically we can test anything we want on the record.
each record is created from what we send to the relevant function (`info`/`debug` etc).
Basically we can add arguments to the logging calls, but it will be really awkward to do in my opinion

![Uploading image.png…]()

Reply:
what do you call an extra variable? basically we can test anything we want on the record.
each record is created from what we send to the relevant function (info/debug etc).
Basically we can add arguments to the logging calls, but it will be really awkward to do in my opinion

Reply:
as said previously this specific implementation wouldn't work, because the logs a re dynamic you can't map them easily, also using a regex can become tricky, since each IaC tool has a different idea on how their resources look like. I think, also what @YaaraVerner said adding the `extra` keyword to all log messages which needs filtering is easier to realize, but it means work to go through all the logs, which we need to do anyway.
ex.
```python
logger.info(f'Failed on resource {resource}', extra={"mask": True})
```

Reply:
2 things here:
1. There is no problem to add this logic to the filter if we decide this is what we want. That's a few lines changed in the `filter` function which can be easily solved.
2. I assumed this is what @YaaraVerner meant, but I just feels it will be extremely tedious to implement. It also means that everybody who use this package should be aware to this feature, which is extremely problematic in my opinion. If we want to do something like that we also need to make sure every `logging.<something>` usage actually suplies this parameter.

Reply:
added now :)